model-optimizer/extensions/middle/DropoutWithRandomUniformReplacer.py

```diff
@@ -0,0 +1,98 @@
+"""
+ Copyright (C) 2018-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import numpy as np
+
+from typing import Dict
+
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, Node, rename_nodes
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.broadcast import Broadcast
+
+
+class DropoutWithRandomUniformReplacer(MiddleReplacementPattern):
```

Comment:
After additional discussion with @jane-intel we decided to move this transformation to TF front phase. This will result in a much smaller pattern and faster solution. Since we get the precision from the Const node we can do this during the front phase.

Reply:
On the front phase it will not work since patterns for the models are different before constant folding.

Reply:
@lazarevevgeny, I moved the transformation to the front, as we agreed.